src/htm/encoders/SimHashDocumentEncoder.cpp

```diff
@@ -56,63 +56,153 @@ namespace htm {
       << "Need only one argument of: 'activeBits' or 'sparsity'.";
     NTA_CHECK(parameters.size > 0u)
       << "Missing 'size' argument.";
+    if ((parameters.tokenFrequencyCeiling > 0) && (parameters.tokenFrequencyFloor > 0)) {
+      NTA_CHECK(parameters.tokenFrequencyCeiling > parameters.tokenFrequencyFloor)
+        << "Argument 'tokenFrequencyCeiling' should be greater than argument 'tokenFrequencyFloor'.";
+    }
+    if (!parameters.tokenSimilarity) {
+      NTA_CHECK(parameters.charFrequencyCeiling == 0)
+        << "Argument 'charFrequencyCeiling' requires argument 'tokenSimilarity'.";
+      NTA_CHECK(parameters.charFrequencyFloor == 0)
+        << "Argument 'charFrequencyFloor' requires argument 'tokenSimilarity'.";
+      if ((parameters.charFrequencyCeiling > 0) && (parameters.charFrequencyFloor > 0)) {
+        NTA_CHECK(parameters.charFrequencyCeiling > parameters.charFrequencyFloor)
+          << "Argument 'charFrequencyCeiling' should be greater than argument 'charFrequencyFloor'.";
+      }
+    }
+    if (!parameters.vocabulary.size()) {
+      NTA_CHECK(!parameters.encodeOrphans)
+        << "Argument 'encodeOrphans' requires argument 'vocabulary'.";
+    }
 
     // Make local copy of arguments for processing
     args_ = parameters;
 
     // process: handle 'sparsity' param instead of 'activeBits' param
-    if(args_.sparsity > 0.0f) {
+    if (args_.sparsity > 0.0f) {
```

Comment:
what if sparsity is set to `-5` ? remove the IF and always check?

Reply:
improved: https://github.com/htm-community/htm.core/pull/603/commits/1d1761159c155851965c0258dba7e8744cf49507